tests/test_hooks.py

```diff
@@ -83,6 +83,54 @@ async def listener(request, response):
     assert resp.body == b'Really this is a bad request'
 
 
+async def test_headers_hook_cannot_consume_request_body(client, app):
+
+    @app.listen('headers')
+    async def listener(request, response):
+        if not request.route.payload:
+            try:
+                request.body
+            except HttpError:
+                response.status = 200
+                response.body = b'raised as expected'
+                return True  # Shortcuts the response process.
+
+    resp = await client.get('/not-found')
+    assert resp.status == HTTPStatus.OK
+    assert resp.body == b'raised as expected'
+
+
+async def test_headers_hook_can_consume_request_body_explicitly(client, app):
+
+    @app.listen('headers')
+    async def listener(request, response):
+        response.status = 200
+        await request.load_body()
```

Comment:
I find a bit ugly that we need two lines to consume a lazy body. My first approach was to write: `response.body = await request.read()`, but 1. there is no `read` method, 2 the `load_body` method does not return the body.

This has been changed [here](https://github.com/pyrates/roll/pull/89/commits/02c40531253d7c2b7676ba422acb489b475b5a75) but I see [here](https://github.com/pyrates/roll/pull/89#discussion_r357960321) that *I* pointed to this change (this shows how inconsistent I am :p )

I'm sort of tempted to revert that commit and have back a `read` method, which I find more natural, and I think it's explicit that a method that return something will but this something in memory.

Thoughts ? :)

Reply:
Can you rewrite `it's explicit that a method that return something will but this something in memory` in English please :p

Reply:
s/but/put/ ! :p

Reply:
If I remember correctly, we wanted to have an explicit read/load step for future usage/flexibility, but this is an old discussion and I'm not sure it's still useful… now that it's released do we want to introduce a breaking change for that? 🤔

Reply:
No need for a breaking change:

```
async def read(self):
    await self.load_body()
    return self._body
```

or

```
async def read(self):
    …

load_body = read
```

Reply:
Indeed, I prefer the first approach, maybe we'll change our mind again at some point in the future 🙈